client/camera.lua

```diff
@@ -136,6 +137,7 @@ end)
 
 CreateThread(function()
 	while true do
+		if not isLoggedIn then return end
```

Comment:
```suggestion
local function handleCamera()
	CreateThread(function()
	while holdingCam do
		lib.requestAnimDict(camanimDict)
	
		if not IsEntityPlayingAnim(cache.ped, camanimDict, camanimName, 3) then
			TaskPlayAnim(cache.ped, camanimDict, camanimName, 1.0, -1, -1, 50, 0, 0, 0, 0)
		end
	
		DisablePlayerFiring(cache.playerId, true)
		DisableControlAction(0,25, true)
		DisableControlAction(0, 44, true)
		DisableControlAction(0,37, true)
		SetCurrentPedWeapon(cache.ped, `WEAPON_UNARMED`, true)
		Wait(100)
	end
end)

RegisterNetEvent('qbx_newsjob:client:toggleCam', function()
    if QBX.PlayerData.job.name ~= 'reporter' then return end
    if not holdingCam then
        lib.requestModel(camModel)
        local plyCoords = GetOffsetFromEntityInWorldCoords(cache.ped, 0.0, 0.0, -5.0)
        local camspawned = CreateObject(camModel, plyCoords.x, plyCoords.y, plyCoords.z, 1, 1, 1)
        Wait(1000)
        local netid = ObjToNet(camspawned)
        SetNetworkIdExistsOnAllMachines(netid, true)
        NetworkSetNetworkIdDynamic(netid, true)
        SetNetworkIdCanMigrate(netid, false)
        AttachEntityToEntity(camspawned, cache.ped, GetPedBoneIndex(cache.ped, 28422), 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 1, 1, 0, 1, 0, 1)
        TaskPlayAnim(cache.ped, camanimDict, camanimName, 1.0, -1, -1, 50, 0, 0, 0, 0)
        cam_net = netid
        holdingCam = true
        lib.showTextUI(Lang:t('info.weazle_overlay'))
        handleCamera()
    else
	    lib.hideTextUI()
        ClearPedSecondaryTask(cache.ped)
        DetachEntity(NetToObj(cam_net), 1, 1)
        DeleteEntity(NetToObj(cam_net))
        cam_net = nil
        holdingCam = false
    end
end)
```

So rather than return this thread early we should delay the thread creation until we actually need it.  Something like this to replace everything in this area. Then do something similar for the other threads where 
they get put them into functions and then call those functions as needed when you activate the camera/mic boom etc.

This will reduce the thread count as well as clean up the readability of this resource.

Reply:
You are correct. Making changes now..

Reply:
Just testing now